src/BigChange.MassTransit.AwsKeyManagementService.Cache/GenerateDataKeyResultExtensions.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using Amazon.Runtime.SharedInterfaces;
+
+namespace BigChange.MassTransit.AwsKeyManagementService.Cache
+{
+    public static class GenerateDataKeyResultExtensions
+    {
+        public static Byte[] ToByteArray(this GenerateDataKeyResult result)
+        {
+            var formatter = new BinaryFormatter();
+            using (var ms = new MemoryStream())
+            {
+                formatter.Serialize(ms, new GenerateDataKeyResultSerializable
+                {
+                    KeyCiphertext = result.KeyCiphertext,
+                    KeyPlaintext = result.KeyPlaintext
+                });
+                return ms.ToArray();
+            }
+        }
+
+        public static GenerateDataKeyResult FromByteArray(this GenerateDataKeyResult result, Byte[] item)
```

Comment:
This seems weird to be an extension method, as I need an instance before I can create another instance from a byte array.